File: 1732.ts
function largestAltitude(gain: number[]): number {
  let sum: number = 0;
  let highestGain: number = 0;

  for (const g of gain) {
    sum += g;
    highestGain = Math.max(sum, highestGain);
  }

  return highestGain;
}

File: 1732.py
from typing import List


class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        largest_altitude = 0
        sum = 0
        
        for net_gain in gain:
            sum += net_gain
            largest_altitude = max(largest_altitude, sum)

        return largest_altitude

sol = Solution()
sol.largestAltitude([-4,-3,-2,-1,4,3,2])


